src/pages/Tags/Tags.tsx

```diff
@@ -1,9 +1,242 @@
+import Container from 'components/atoms/Container/Container'
+import DynamicForm, {
+  FieldProps,
+  InputTypes,
+} from 'components/organisms/DynamicForm/DynamicForm'
+import { ReactComponent as Add } from 'assets/icons/add.svg'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import Button, {
+  AppearanceTypes,
+  IconPositioningTypes,
+} from 'components/molecules/Button/Button'
+import { useFetchTags } from 'hooks/requests/useTags'
+import { useBulkCreate } from 'hooks/requests/useTags'
+import { filter, flatMap, groupBy, includes, map, uniqBy } from 'lodash'
+import { ReactComponent as EditIcon } from 'assets/icons/edit.svg'
+import { TagsType } from 'types/tags'
+import { showNotification } from 'components/organisms/NotificationRoot/NotificationRoot'
+import { NotificationTypes } from 'components/molecules/Notification/Notification'
+import Loader from 'components/atoms/Loader/Loader'
+import { v4 as uuidv4 } from 'uuid'
+import useAuth from 'hooks/useAuth'
+import { Privileges } from 'types/privileges'
+import useValidators from 'hooks/useValidators'
+import { showValidationErrorMessage } from 'api/errorHandler'
 
-// TODO: WIP - implement this page
+import classes from './classes.module.scss'
+
+export interface ObjectType {
+  [key: string]: string
+}
+
+export type FormValues = {
+  tagInput?: ObjectType
+  tagCategorySelection?: string
+}
 
 const Tags: FC = () => {
-  return <h1>Tags</h1>
+  const { t } = useTranslation()
+  const { tagInputValidator } = useValidators()
+  const { userPrivileges } = useAuth()
+
+  const { tags, isLoading: isFetchingTags } = useFetchTags()
+  const { createTags, isLoading: isCreatingTags } = useBulkCreate()
+
+  const tagCategoryOptions = map(tags, ({ type }) => {
+    return {
+      label: type || '',
+      value: type || '',
+    }
+  })
+
+  const uniqueTagCategoryOptions = uniqBy(tagCategoryOptions, 'label')
+
+  const updatedDataWithoutSkillsObject = filter(
+    uniqueTagCategoryOptions,
+    (categoryObjects) => categoryObjects.label !== 'Oskused'
+  )
+
+  const {
+    control,
+    handleSubmit,
+    formState: { dirtyFields },
+    reset,
+  } = useForm<FormValues>({
+    reValidateMode: 'onSubmit',
+  })
+
+  const tagFields: FieldProps<FormValues>[] = [
+    {
+      inputType: InputTypes.Text,
+      ariaLabel: t('tag.tag_name'),
+      label: 'Nimetus',
+      name: `tagInput.${uuidv4()}`,
+      placeholder: t('tag.tag_input'),
+      type: 'text',
+      rules: {
+        required: true,
+        validate: tagInputValidator,
+      },
+      className: classes.tagInputField,
+      disabled: !includes(userPrivileges, Privileges.AddTag),
+    },
+  ]
+
+  const categoryFields: FieldProps<FormValues>[] = [
+    {
+      inputType: InputTypes.Selections,
+      name: 'tagCategorySelection',
+      ariaLabel: t('tag.select_tag_category'),
+      options: updatedDataWithoutSkillsObject,
+      placeholder: t('tag.select_tag_category'),
+      multiple: true,
```

Comment:
**[blocker]**

1. According to wiki this should be a single category input. Tags are added to 1 category at a time.
2. This means we also don't need the "buttons: true" here